[AMD] Turn on fast path for index_put

xw285cornell opened the pull request:
Summary:
This slow path is bad because it has a sync point which makes CPU really slow. I'm not very sure if AMD actually needs this with the newer rocm versino

{F1870213925}

Test Plan: CI

Differential Revision: D62731130

danzimm reviewed (approved):
LGTM

danzimm commented:
@jeffdaily does the [ciflow/rocm](https://github.com/pytorch/pytorch/labels/ciflow%2Frocm) label trigger some additional AMD specific CI?

jeffdaily commented:
> @jeffdaily does the [ciflow/rocm](https://github.com/pytorch/pytorch/labels/ciflow%2Frocm) label trigger some additional AMD specific CI?

We have a limited pool of CI servers.  The ciflow/trunk will trigger a fraction of the full CI suite (no inductor, a select set of test modules, etc).  OTOH, ciflow/rocm runs much more.  We try to limit using ciflow/rocm on PRs because it puts pressure on our CI resources.  We have the authority to revert merged PRs if they break ROCm, in which case we add the ciflow/rocm label to the reverted PR to aid the PR author in fixing the issue for ROCm.

xw285cornell committed:
[AMD] Turn on fast path for index_put (#136136)

Summary:

This slow path is bad because it has a sync point which makes CPU really slow. I'm not very sure if AMD actually needs this with the newer rocm versino

{F1870213925}

Test Plan: CI

Reviewed By: danzimm

Differential Revision: D62731130
